src/vivarium_gates_nutrition_optimization/components/observers.py

```diff
@@ -180,8 +180,8 @@ def setup(self, builder: Builder) -> None:
         for intervention in models.SUPPLEMENTATION_CATEGORIES:
             builder.results.register_observation(
                 name=f"intervention_{intervention}_count",
-                pop_filter=f'alive == "alive" and intervention == "{intervention}" and tracked == True',
-                requires_columns=["alive", "intervention"],
+                pop_filter=f'alive == "alive" and intervention == "{intervention}" and on_treatment == True and previous_on_treatment == False and tracked == True',
+                requires_columns=["alive", "intervention","on_treatment","previous_on_treatment"],
```

Comment:
Since nobody dies and everyone is getting put on treatment at the same time the `on_treatment` columns are not providing any more information than just the clock. You should be able to replace this pop_filter with 
```python
intervention_date = self.start_date + timedelta(days=data_values.DURATIONS.INTERVENTION_DELAY_DAYS)
pop_filter=(
    f'alive == "alive and {intervention_date} < event_time'
    f' and event_time <= {intervention_date + self.time_step}"
)
```

Then you don't actually need any new columns.

Reply:
ahhh, I forgot (or perhaps never knew) that you can just directly access the event time. I wanted to do something like this but didn't know how. Yes, this is much better.

Reply:
done